### ISSUE TITLE
Server crashes with java.lang.ClassNotFoundException for ir.mehradn.cavesurvey.util.upgrades.CaveMapUpgrade

### ISSUE THREAD

**Issue by Ghifari160** [MehradN/CaveSurvey#1]
**Minecraft Version:** 1.19.4
**Fabric Loader Version:** 0.14.19
**Operating System:** Windows 10
**Java Version:**
```
java version "17.0.2" 2022-01-18 LTS
Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.2+8-LTS-86, mixed mode, sharing)
```

The server version of CaveSurvey instantly crashes on launch with nothing else other than the mod and the Fabric API installed. From the stack trace, it looks like `CaveMapUpgrade` class doesn't exist or otherwise can't be found by the JVM or Fabric. I've attached [latest.log](https://github.com/MehradN/CaveSurvey/files/11338837/latest.log).

The same jar works fine when installed on the client.

**Comment by Ghifari160** [comment]
From a glance, it looks like the [`ir.mehradn.cavesurvey.util.upgrades.ServerCaveMapUpgrade`](https://github.com/MehradN/CaveSurvey/blob/master/server/src/main/java/ir/mehradn/cavesurvey/util/upgrades/ServerCaveMapUpgrade.java#L9) extends `CaveMapUpgrade` which doesn't exist under the same package on the server codebase. [`ir.mehradn.cavesurvey.util.upgrades.ClientCaveMapUpgrade`](https://github.com/MehradN/CaveSurvey/blob/master/client/src/main/java/ir/mehradn/cavesurvey/util/upgrades/ClientCaveMapUpgrade.java#L17) also extends `CaveMapUpgrade`. But for the client, that class is defined [in the package](https://github.com/MehradN/CaveSurvey/blob/master/client/src/main/java/ir/mehradn/cavesurvey/util/upgrades/CaveMapUpgrade.java). This might need more than a simple copy-paste between the two directories as `CaveMapUpgrade` refers to [`ir.mehradn.cavesurvey.util.CaveMapTagManager`](https://github.com/MehradN/CaveSurvey/blob/master/client/src/main/java/ir/mehradn/cavesurvey/util/CaveMapTagManager.java), which might have some client-specific code.

As a slight fyi: It might be worth merging the two code bases and just declaring the client-specific code and server-specific code with `@Environment` annotations and through the mixin config in `fabric.mod.json`. I don't know how much of this is worth doing though. It's a lot or reorganization, but it might help your sanity down the line if you plan on keeping up with the mod. Also take this part with a grain of salt as I'm not too familiar with modding with Fabric (I have tried modding through Forge a couple of times, and the concept of sided-code seems similar enough).

**Comment by MehradN** [comment]
The reason I'm keeping these as separate jars is for the situation where you play in a server that has cave survey installed, but you don't want to install cave survey on your single player worlds. I could either split the project to two jars (which I did) or I could add some configuration to disable the mod, which would make setting up the mod more annoying and complicated as it already is. Most likely I won't be merging the jar files any time soon.

Because of this strange behavior where the mod works fine on clients, and it works fine on the server debugging, but it doesn't work on actual servers, the issue is either because I've accidentally used a client class inside the CaveMapUpgrade or I've messed up the build.gradle. I'm investigating the first possibility but if it's the second possibility then I'm going to need help from people who have knowledge in fabric and gradle.

Hopefully the problem will be fixed until tomorrow.

**Comment by MehradN** [comment]
Solved by de154fd!

The problem was way more basic than I thought. In fabric mod properties there is a property named "environment" which allows you to specify if the mod is client or server only. Only and only on actual fabric servers there is this feature that these mods don't get loaded by the fabric loader, at all.

The problem is fixed now and the mod should work fine on any platform.

Thank you for reporting the issue and I hope you have a wonderful time!
